### ISSUE TITLE
Generate Julia types from XML schema files

### ISSUE THREAD

**Issue by kskyten** [aminya/AcuteML.jl#154]
It would be extremely useful to generate Julia types from XML schema files automatically. This would make it easy to generate, for example, SVG and MathML.  [Here](https://github.com/simonpoulding/DataGeneratorTranslators.jl/tree/master/src/xsd) is some useful code to handle XSD fschema files in Julia.

**Comment by aminya** [comment]
You can parse any XML file if you declare its structure using `@aml` macro.

**Comment by kskyten** [comment]
Yes, but you have to define the structure of the XML in Julia first. What I'm looking for is automatically generating this Julia code to parse and generate XML based on it's schema. In MusicXML's case I would take the XML schema from here: https://www.musicxml.com/for-developers/musicxml-xsd/ and automatically generate the AcuteML code.

**Comment by aminya** [comment]
@kskyten This seems like a very good idea. However, I am not sure how we should approach this.

It is also related to #93.

**Comment by el-oso** [comment]
Hmmm... Actually I would also benefit from this functionality. I have some XML Schemas that are generated by other people and would be great to automatically create readers and writers based on those.

**Comment by goretkin** [comment]
What would the different types enable you to do, as opposed to using a `Dict` or `NamedTuple`?

Note for example that Julia's own AST representation for its syntax uses (for non-leaf nodes) the same type `Expr`.
Or another example: https://github.com/yurivish/Hyperscript.jl/blob/c780f6dcd95fbf8931c7afc00499a8436075ae6c/src/Hyperscript.jl#L79-L84
